src/ReverseProxy/Abstractions/Config/ITransformFactory.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+
+namespace Microsoft.ReverseProxy.Abstractions.Config
+{
+    /// <summary>
+    /// Used to validate and build transforms from the given parameters.
+    /// </summary>
+    public interface ITransformFactory
+    {
+        /// <summary>
+        /// Checks if the given transform values match a known transform, and if those values have any errors.
+        /// </summary>
+        /// <param name="context">The context to add any generated errors to.</param>
+        /// <param name="transformValues">The transform values to validate.</param>
+        /// <returns>True if this factory matches the given transform, otherwise false.</returns>
+        public bool Validate(TransformValidationContext context, IReadOnlyDictionary<string, string> transformValues);
```

Comment:
Is this intentional?
```suggestion
        bool Validate(TransformValidationContext context, IReadOnlyDictionary<string, string> transformValues);
```

Reply:
Hmm, that used to cause a compiler error... Maybe they relaxed that in C#9 or one of the new compilers?

Removed.